pcr-signature-generator.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+set -euo pipefail
+
+# If GRUB2 is used, bli.mod needs to be loaded
+EFIVAR="/sys/firmware/efi/efivars/LoaderDevicePartUUID-4a67b082-0a4c-41cf-b6c7-440b29bb8c4f"
+
+[ -e "$EFIVAR" ] || exit 0
+
+# Read the value of the EFI variable, that contains a header and ends
+# with '\0' and make it lowercase
+ESP_UUID="$(dd "if=$EFIVAR" bs=2 skip=2 conv=lcase status=none | tr -d '\0')"
+DEV="/dev/disk/by-partuuid/${ESP_UUID}"
+
+cat > "$1/boot-efi.mount" <<EOF
```

Comment:
Most generators use `/tmp` if $1 is not set. Currently it would fail with an undefined variable error

Reply:
(Ironically using `/tmp` is actually a security issue - it touches predictable paths in a world-writable directory)

Reply:
Interesting, according to https://www.freedesktop.org/software/systemd/man/latest/systemd.generator.html this first parameter is not optional. I can fail if it is not set, that would make sense

Reply:
Checking presence of $1. Done